Source/NSubstitute/Core/CallSpecificationFactory.cs

```diff
@@ -15,7 +15,7 @@ public ICallSpecification CreateFrom(ICall call, MatchArgs matchArgs)
         {
             var methodInfo = call.GetMethodInfo();
             var argumentSpecs = call.GetArgumentSpecifications();
-            var arguments = call.GetArguments();
+            var arguments = call.GetOriginalArguments();
```

Comment:
Hi @Zvirja, can you show an example test that will fail if this line stayed as `call.GetArguments()`?